jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ContextHandler.java

```diff
@@ -228,6 +229,12 @@ public ContextHandler(String contextPath)
         this(null, null, contextPath);
     }
 
+    public ContextHandler(String contextPath, org.eclipse.jetty.ee9.nested.Handler handler)
```

Comment:
We need some javadoc somewhere that describes how this Context can be invoked in 2 different ways:
 1. via it's nested coreContextHandler which will wrap the core request/response and create the nest baseRequest/baseResponse
 2. via a direct call to `handle` when this handler is nested within another context handler.

In fact, I really don't like it that this class can simultaneously be used in both ways. In the case of a nested contextHandler, that means there is a second coreContextHandler just doing nothing.       I'd prefer to have a mode where the second coreContextHandler was never created and only then could it receive requests from handle.

At the very least, this needs to be javadoc'd, but I'd prefer a mode.... I don't think splitting is possible.

Perhaps the mode can be established by a call to the get of the coreContextHandler?

Reply:
I think the split is possible. Maybe if the `CoreContextHandler` had a nested `ContextHandler` and delegated to that instead of the other way around. 

Right now nested `ContextHandler` delegates lots of its methods to `CoreContextHandler` and its `ScopedContext`. So hard to separate them without some more fundamental changes.

Anyway this would be for another PR if we decide to do it. For now I have added some javadoc to explain the two different cases how this can be invoked.